src/gui_common/art_gallery/GalleryCard.cs

```diff
@@ -44,6 +52,18 @@ public override void _Ready()
         UpdatePreview();
     }
 
+    public override void _Process(float delta)
+    {
+        base._Process(delta);
+
+        time += delta;
+        if (IsImageOnlyCard && !isLoaded && time >= CardIndex * LOAD_TIME_PER_CARD)
```

Comment:
How does this prevent multiple cards from loading at the same time?
I was thinking that a new ResourceManager class would make sense to create, it would work similarly to how the photostudio works in that you submit resource loading tasks to it, and then it loads one thing per frame (or even waits between frames to not tank FPS below 30 or so).